asakusafw/asakusafw
# Enable Direct I/O `DataFormat` for test inputs.

## Summary

This PR enables Direct I/O `DataFormat` for test inputs.

## Background, Problem or Goal of the patch

N/A.

## Design of the fix, or a new feature

This introduces the following new methods:

* `{FlowPart,JobFlow}Driver{Input,Output}<T>`
  * `prepare(Class<? extends DataFormat<? super T>> formatClass, File sourceFile)`
    * load the given file
  * `prepare(Class<? extends DataFormat<? super T>> formatClass, String sourcePath)`
    * load a file on the classpath (relative from the test class)
* `OperatorTestEnvironment`
  * `loader(Class<T> dataType, Class<? extends DataFormat<? super T>> formatClass, File file)`
    * load the given file
  * `loader(Class<T> dataType, Class<? extends DataFormat<? super T>> formatClass, String path)`
    * load a file on the classpath (relative from the test class)

Note that, the "file name" may be changed if the given source path is type of `String`, so that use `File` instead if file path affects the tests (e.g. the data format has `@directio.*.file_name` field).

## Related Issue, Pull Request or Code

* #700 (`OperatorTestEnvironment.loader()`)

## Wanted reviewer

@akirakw
